es-app/src/FileData.cpp

```diff
@@ -134,12 +126,27 @@ std::vector<FileData*> FileData::getFilesRecursive(unsigned int typeMask, bool d
 	return out;
 }
 
+std::string FileData::getKey() {
+	if (mIsVirtual)
+		return getLongKey();
+	else
+		return getShortKey();
+}
+
+FileData* FileData::getMetadataSourceFileData()
```

Comment:
Well... Your comment actually makes sense. From a client point of view, you are right. They don't care. They want to get the original instance of the File being manipulated - use cases are, usually, for metadata editing, for file deletion, etc.

In that sense, I wouldn't be shocked with renaming it to something more transparent. I just associate "get()" with "getting the actual instance". Would "getSource()" be better? "getFileData()" (given that, for FileData objects it returns "this", but for "CollectionFileData" objects it returns the FileData object it's cloned from)?

Thanks!